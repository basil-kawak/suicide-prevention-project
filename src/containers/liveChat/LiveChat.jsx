import React from "react";
import ChatBot from "react-simple-chatbot";
import { ThemeProvider } from "styled-components";

const THEME = {
  background: "#f5f8fb",
  fontFamily: "Helvetica Neue",
  headerBgColor: "#539be5",
  headerFontColor: "#fff",
  headerFontSize: "15px",
  botBubbleColor: "#539be5",
  botFontColor: "#fff",
  userBubbleColor: "#fff",
  userFontColor: "#4a4a4a",
  floating: true,
};

const STEPS = [
  {
    id: "1",
    message: "What is your name?",
    trigger: "2",
  },
  {
    id: "2",
    user: true,
    trigger: "3",
  },
  {
    id: "3",
    message: "Hi {previousValue}, nice to meet you!",
    trigger: "4",
  },
  {
    id: "4",
    message: "What number I am thinking?",
    trigger: "5",
  },
  {
    id: "5",
    options: [
      { value: 1, label: "Number 1", trigger: "7" },
      { value: 2, label: "Number 2", trigger: "6" },
      { value: 3, label: "Number 3", trigger: "6" },
    ],
  },
  {
    id: "6",
    message: "Wrong answer, try again.",
    trigger: "5",
  },
  {
    id: "7",
    message: "Awesome! You are a telepath!",
    end: true,
  },
];

const LiveChat = () => {
  return (
    <>
      <ThemeProvider theme={THEME}>
        <ChatBot
          steps={STEPS}
          floating={true}
          headerTitle="Speech Recognition"
          recognitionEnable={true}
        />
      </ThemeProvider>
    </>
  );
};

export default LiveChat;
