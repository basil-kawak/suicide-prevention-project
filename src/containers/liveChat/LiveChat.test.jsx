import React from "react";
import LiveChat from "./LiveChat";
import { render } from "@testing-library/react";

it("blog display snapshot", () => {
  const tree = render(<LiveChat />);
  expect(tree).toMatchSnapshot();
});
