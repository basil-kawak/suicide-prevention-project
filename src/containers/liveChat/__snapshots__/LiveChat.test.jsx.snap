// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blog display snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rsc "
      >
        <a
          class="sc-fznZeY gHaEAM rsc-float-button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 2H4c-1.1 0-2 .9-2 2v18l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </a>
        <div
          class="sc-fzoLsD bAaTLk rsc-container"
          height="520px"
          width="350px"
        >
          <div
            class="sc-fzplWN cetEoJ rsc-header"
          >
            <h2
              class="sc-fznyAO hCdxiT rsc-header-title"
            >
              Speech Recognition
            </h2>
            <a
              class="sc-fznKkj kqtOvC rsc-header-close-button"
            >
              <svg
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </a>
          </div>
          <div
            class="sc-fzpans dgsAPV rsc-content"
            height="520px"
          >
            <div
              class="sc-fzozJi gOCww rsc-ts rsc-ts-bot"
            >
              <div
                class="sc-AxmLO jmCkbN rsc-ts-image-container"
              >
                <img
                  alt="avatar"
                  class="sc-Axmtr dJJgye rsc-ts-image"
                  src="data:image/svg+xml,%3csvg version='1' xmlns='http://www.w3.org/2000/svg' viewBox='0 0 512 512'%3e%3cpath d='M303 70a47 47 0 1 0-70 40v84h46v-84c14-8 24-23 24-40z' fill='%2393c7ef'/%3e%3cpath d='M256 23v171h23v-84a47 47 0 0 0-23-87z' fill='%235a8bb0'/%3e%3cpath fill='%2393c7ef' d='M0 240h248v124H0z'/%3e%3cpath fill='%235a8bb0' d='M264 240h248v124H264z'/%3e%3cpath fill='%2393c7ef' d='M186 365h140v124H186z'/%3e%3cpath fill='%235a8bb0' d='M256 365h70v124h-70z'/%3e%3cpath fill='%23cce9f9' d='M47 163h419v279H47z'/%3e%3cpath fill='%2393c7ef' d='M256 163h209v279H256z'/%3e%3cpath d='M194 272a31 31 0 0 1-62 0c0-18 14-32 31-32s31 14 31 32z' fill='%233c5d76'/%3e%3cpath d='M380 272a31 31 0 0 1-62 0c0-18 14-32 31-32s31 14 31 32z' fill='%231e2e3b'/%3e%3cpath d='M186 349a70 70 0 1 0 140 0H186z' fill='%233c5d76'/%3e%3cpath d='M256 349v70c39 0 70-31 70-70h-70z' fill='%231e2e3b'/%3e%3c/svg%3e"
                />
              </div>
              <div
                class="sc-AxheI kTPiBC rsc-ts-bubble"
              >
                <span
                  class="rsc-loading"
                >
                  <span
                    class="sc-AxjAm gvbnAi"
                  >
                    .
                  </span>
                  <span
                    class="sc-AxjAm ia-DwmC"
                  >
                    .
                  </span>
                  <span
                    class="sc-AxjAm hZHQYI"
                  >
                    .
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div
            class="sc-fzqBZW gsytLs rsc-footer"
          >
            <input
              class="sc-fzqNJr cxzqCi rsc-input"
              disabled=""
              placeholder="Type the message ..."
              type="textarea"
              value=""
            />
            <button
              class="sc-fzoyAV cyKRBF rsc-submit-button"
              disabled=""
            >
              <svg
                height="20"
                version="1.1"
                viewBox="0 0 500 500"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <g>
                    <polygon
                      points="0,497.25 535.5,267.75 0,38.25 0,216.75 382.5,267.75 0,318.75"
                    />
                  </g>
                </g>
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rsc "
    >
      <a
        class="sc-fznZeY gHaEAM rsc-float-button"
      >
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 2H4c-1.1 0-2 .9-2 2v18l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </a>
      <div
        class="sc-fzoLsD bAaTLk rsc-container"
        height="520px"
        width="350px"
      >
        <div
          class="sc-fzplWN cetEoJ rsc-header"
        >
          <h2
            class="sc-fznyAO hCdxiT rsc-header-title"
          >
            Speech Recognition
          </h2>
          <a
            class="sc-fznKkj kqtOvC rsc-header-close-button"
          >
            <svg
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </a>
        </div>
        <div
          class="sc-fzpans dgsAPV rsc-content"
          height="520px"
        >
          <div
            class="sc-fzozJi gOCww rsc-ts rsc-ts-bot"
          >
            <div
              class="sc-AxmLO jmCkbN rsc-ts-image-container"
            >
              <img
                alt="avatar"
                class="sc-Axmtr dJJgye rsc-ts-image"
                src="data:image/svg+xml,%3csvg version='1' xmlns='http://www.w3.org/2000/svg' viewBox='0 0 512 512'%3e%3cpath d='M303 70a47 47 0 1 0-70 40v84h46v-84c14-8 24-23 24-40z' fill='%2393c7ef'/%3e%3cpath d='M256 23v171h23v-84a47 47 0 0 0-23-87z' fill='%235a8bb0'/%3e%3cpath fill='%2393c7ef' d='M0 240h248v124H0z'/%3e%3cpath fill='%235a8bb0' d='M264 240h248v124H264z'/%3e%3cpath fill='%2393c7ef' d='M186 365h140v124H186z'/%3e%3cpath fill='%235a8bb0' d='M256 365h70v124h-70z'/%3e%3cpath fill='%23cce9f9' d='M47 163h419v279H47z'/%3e%3cpath fill='%2393c7ef' d='M256 163h209v279H256z'/%3e%3cpath d='M194 272a31 31 0 0 1-62 0c0-18 14-32 31-32s31 14 31 32z' fill='%233c5d76'/%3e%3cpath d='M380 272a31 31 0 0 1-62 0c0-18 14-32 31-32s31 14 31 32z' fill='%231e2e3b'/%3e%3cpath d='M186 349a70 70 0 1 0 140 0H186z' fill='%233c5d76'/%3e%3cpath d='M256 349v70c39 0 70-31 70-70h-70z' fill='%231e2e3b'/%3e%3c/svg%3e"
              />
            </div>
            <div
              class="sc-AxheI kTPiBC rsc-ts-bubble"
            >
              <span
                class="rsc-loading"
              >
                <span
                  class="sc-AxjAm gvbnAi"
                >
                  .
                </span>
                <span
                  class="sc-AxjAm ia-DwmC"
                >
                  .
                </span>
                <span
                  class="sc-AxjAm hZHQYI"
                >
                  .
                </span>
              </span>
            </div>
          </div>
        </div>
        <div
          class="sc-fzqBZW gsytLs rsc-footer"
        >
          <input
            class="sc-fzqNJr cxzqCi rsc-input"
            disabled=""
            placeholder="Type the message ..."
            type="textarea"
            value=""
          />
          <button
            class="sc-fzoyAV cyKRBF rsc-submit-button"
            disabled=""
          >
            <svg
              height="20"
              version="1.1"
              viewBox="0 0 500 500"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <g>
                  <polygon
                    points="0,497.25 535.5,267.75 0,38.25 0,216.75 382.5,267.75 0,318.75"
                  />
                </g>
              </g>
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
